### PR TITLE
[sparkle] - enh: `DataTable`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at enhancing the `DataTable` component, which includes:
- change row opacity on hover
- pass default column ordering
- fix storybook

## Risk

Breaking UI

## Deploy Plan

- Deploy sparkle
- Bump sparkle version in `front` in another PR

### CODE REVIEW

File: sparkle/src/components/DataTable.tsx [hunk 1]
```diff
@@ -197,7 +202,9 @@ DataTable.Row = function Row({
     <tr
       className={classNames(
         "s-border-b s-border-structure-200 s-text-sm",
-        onClick ? "s-cursor-pointer" : "",
+        onClick
+          ? "s-cursor-pointer hover:s-border-b-neutral-500 hover:s-opacity-70"
```
Comment (albandum) [inline sparkle/src/components/DataTable.tsx:206]: Maybe let's only do the border, opacity drop looks a little odd no?
The best would be to add a light grey background to the child <td>'s

Comment (JulesBelveze) [reply sparkle/src/components/DataTable.tsx]: @albandum something like this?

Comment (albandum) [reply sparkle/src/components/DataTable.tsx]: Much better - no need for the border change anymore
Remove the border change and mergeable 👍

**Review by albandum — APPROVED** [review]
Address border change comment and go 👍

**Pull request merged**